### PR TITLE
Triggers

### PR DESCRIPTION
**Pull request by ramessesii2**
added  triggers
* added tekton resources for clone-apply pipeline
* added triggers to auto deploy on a new commit to the 
* added triggers to run tests
* added run.sh to prepare the cluster for mks-server

### How to test
1. Setup the OC cluster and Install OSPipelines
2. Run the script to apply dependencies ```./scripts/run.sh```
3. To test pipelineruns-
              i. create the pipelineruns resources using ```oc create -f <pipelinerun>```
              ii. check the logs using ```tkn pipelinerun logs -f```
4. To expose the trigger services-
              i. ``` oc expose svc el-<trigger-service-name>```
              ii. Add url to github webhook - 
                   ```echo "$(oc  get route el-<trigger-service-name> --template='http://{{.spec.host}}')"```

### CODE REVIEW

File: k8s/mks-config/clusterRoleMKS.yaml [hunk 1]
```diff
@@ -22,13 +22,17 @@ metadata:
   labels:
     app.kubernetes.io/instance: mks-server-clusterRole
 rules:
+  - apiGroups: ["mkscontroller.example.mks"]
+    resources: ["mkstasks", "mkstaskruns", "mkspipelineruns"]
```
Comment (avinal) [inline k8s/mks-config/clusterRoleMKS.yaml]:
```suggestion
    resources: ["mkstasks", "mkstaskruns", "mkspipelineruns, "mkspipelines"]
```

File: scripts/run.sh [hunk 2]
```diff
@@ -0,0 +1,64 @@
+#!/bin/sh
```
Comment (avinal) [inline scripts/run.sh:1]: add copyright

Comment (ramessesii2) [reply scripts/run.sh:1]: done

**Review by UtkarshChaurasia — APPROVED** [review]
lgtm

**Pull request merged**